core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -77,18 +76,29 @@ namespace qana {
 
 typedef std::pair<std::string, std::string> StringPair;
 
-std::shared_ptr<query::ColumnRef>
-resolveAsColumnRef(query::QueryContext& context, query::ValueExprPtr vexpr) {
-    std::shared_ptr<query::ColumnRef> cr = vexpr->copyAsColumnRef();
+/// Returns a list of column references.
+/// This class will usually check the list if any can be used
+/// to restrict the query to a subset of worker nodes via the secondary index.
+/// If there is ambiguity in the possible table that a column belongs to,
+/// the SQL statement should cause an error when tried on the local database,
+/// so that check can be skipped.
+///
+query::ColumnRef::Vector resolveAsColumnRef(query::QueryContext& context, query::ValueExprPtr vexpr) {
+    query::ColumnRef::Vector columnRefs;
+    query::ColumnRef::Ptr cr = vexpr->copyAsColumnRef();
     if (!cr) {
-        return cr;
+        return columnRefs;
     }
-    query::DbTablePair p = context.resolve(cr);
-    cr->table = p.table;
-    cr->db = p.db;
-    return cr;
+    std::string column = cr->column;
+    query::DbTableSet set = context.resolve(cr);
+    for (auto const& dbTblPair : set) {
+        cr = std::make_shared<query::ColumnRef>(dbTblPair.db, dbTblPair.table, column);
+        columnRefs.push_back(cr);
```

Comment:
You could do it too:

    columnRefs.emplace_back(new query::ColumnRef(dbTblPair.db, dbTblPair.table, column));

but it's not very exception-safe, object will leak if `emplace_back` throws.